### DISCUSSION TITLE
filtering the tsv if input library is merged from 2 DIA-NN libs

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#620] [Q&A]
Dear Vadim, I hope you are well. 

### Generic case: 
DIA-NN search A generates Library A. DIA-NN search B generates Library B. Of course, MBR is applied to both of them, yielding Lib.PG.Q.Values for A and B. 
I then use Library A and B together to execute search C. 
Q1: How do I need to filter the report? Take the first pass tsv and filter based on Lib.PG.Q.Value < 1%?
Q2: Would this cause an FDR problem? (because Lib.PG.Q.Value doesn't apply any more when combining multiple libraries)
Q3: Or do I have to execute MBR again, i.e. take the second pass tsv and filter based on Global.PG.Q.Value <1%? (isn't this too stringent as MBR has been applied to both individual libraries already)?

### Specific case: 
A, B, C samples are the same matrix, e.g. the same body fluid. All runs are DIA.
samples for A are a subset (<10%) of the actual full set of samples C. I do this, because search A is time-consuming. I generate a restricted library with which I then search the full set C. (Here I would normally use Lib.PG.Q.Val 1% to filter the first pass output)

Now I want to additionally incorporate B:
B samples are another subset of C samples (<10% of C) that I have additionally measured a second time but with a longer gradient. The reason is to enrich the overall library with more identifications that are realistically obtainable in such body fluid. Longer gradient may sound weird at first, but its somewhat similar to depletion or fractionation approaches that also aim to enrich the library with identifications. 
Since it is the same body fluid matrix, there is no problem for searching the B library (long gradient) on C samples (short gradient), as DIANN manages the RT recalibration. 
When I combine the A library (short gradient) with the B library (long gradient), I use the overlapping ions to scale the RT of B (long gradient) down to A (short gradient). This works well and yields a more complete and better library AB.  Searching C now with the AB library provides a benefit in any combination (first pass vs second pass results // Lib.PG.Q.Value vs Global.PG.Q.Value 1% filter) - and the differences between these filter combinations are not huge. Nevertheless it's not clear to me which filtering is the appropriate one to use. 
Q4: does the specific case change anything about the answer for the generic case?

**Comment by vdemichev** [comment]
Hi Jakob,

Two options: generate libraries A and B using IDs profiling only and 1% precursor FDR, or using IDs, RT and IM profiling. In either case, check the RT scales are the same in the two libs, using shared IDs (good even if gradients are the same). If not, align them in R using loess or sth similar. In either case, combine the libraries in a separate pipeline step to obtain a merged .tsv library. 

First case: in the combined lib, delete all columns with q-values. Run search C with MBR. Filter by Lib.PG.Q.Value at 1%.

Second case: run search C without MBR, filter precursors that are not shared between libs by Lib.Q.Value 0.5%, filter proteins that are shared between libs by Lib.PG.Q.Value 1%, filter proteins that are not shared between libs by Lib.PG.Q.Value 0.5%. 

In either case, you are not strictly guaranteed correct FDR control, but it will not inflate signficantly, so all fine. I would go with method one as an easier one.

Btw, can combine this way results with public libs & project-specific libs & lib-free.

Best,
Vadim

**Reply by vdemichev** [reply]
Btw, libs generated with IDs, RT and IM profiling can be easily made equivalent to libs generated with 'IDs profiling' by replacing RTs and IMs with original ones or, if lib-free, using DIA-NN deep learning feature to replace stuff in the lib. So no need to run separate lib generation procedures.

**Reply by vdemichev** [reply]
Another comment: IM values need likewise to be aligned between libs.
Also: can generate a lib from both A and B.

**Reply by JM-Bader** [reply]
Hi Vadim, 
thanks for the swift reply!

FYI: When merging library A and B, we did basically what you suggest: align RT using loess regression. In fact, library A has IM. But B doesn't (it's cross platform). We predict IM and it works well. This all is a separate step outside DIA-NN. We deleted all q-value columns. Only these columns remain: 
![image](https://user-images.githubusercontent.com/98888664/218196018-581fb888-e3bb-411e-a3b2-ddb393ce56af.png)

I have generated libraries A and B as part of 'normal' DIA-NN searches using the default 'smart profiling'. 
Previously, I had generally tested ID, RT, and IM  profiling and have observed little difference (in the main tsv at least). 

Q1: is it an issue if I used smart profiling to generate lib A and B? Why?
Q2: Generating lib A takes quite long, so I would prefer not to repeat. IF Q1 is an issue, is there another way?

Independently, I am sorry - but a bit confused about how the two 'options' (IDs profiling & 1% precursor FDR// IDs, RT, and IM profiling) connect to first and second case (in merged lib, delete all columns with q-values. Run with MBR. Lib.PG.Q.Val 1% filter // running without MBR)
Q3: Can you clarify?

Q4: Just to understand: merged lib: all q-value columns deleted, Run C with MBR. It should be ok to filter by Lib.PG.Q.Value at 1%. However, if I wanted I could instead filter by Global.PG.Q.Value 1% which is bit more conservative. Correct? (These 2 would be my preferred options while keeping the previous searches A and B with 'smart profiling')

Thanks!

**Reply by JM-Bader** [reply]
Also thanks for the btw information - I wasn't aware and built the libraries in a modular way, just had the merging idea very recently.

**Reply by JM-Bader** [reply]
Also FYI: 
there is not such a large difference between the filterings: 
see the cumulative data completeness curve (y = cumulative proteins identified in at least x samples, x axis = x samples)
"_S_" = second pass, "_F_" = first pass
prot = proteins
end of legend = filter criteria (1%)
![image](https://user-images.githubusercontent.com/98888664/218199686-cd735e3d-95d4-4b31-81a9-18f76a0e20f6.png)

**Reply by JM-Bader** [reply]
Q5: is the Lib.PG.Q.Value newly calculated ? Until now, I thought it would be the Global.PG.Q.value of the search that generates the library. But when this value is deleted from the library (as we did), a Lib.PG.Q.value is still present in the report tsv....

**Reply by vdemichev** [reply]
- If one lib has predicted IM and another empirical, better to go for the first method. Otherwise, even if aligning IM between libs, many precursors with predicted IM will be discarded simply because they don't fall within the IM window due to prediction inaccuracy.

- Yes, smart profiling is about the same. If have doubts, can just replace spectra with predicted ones (predict a lib and replace spectra in R).

- I meant first case = first option, second  = second.

- Yes, 'global' q-values are more conservative, unless you delete q-value columns, in which case it's unpredictable.

- The search that generate the lib produces global q-values which are then saved to the lib as 'lib' q-values.

**Reply by JM-Bader** [reply]
Thanks! All clear but the last one. 
I understand your bullet point. But if I edit the library, removing all q-value columns, for the procedure above - then there should not be a Lib.PG.Q.value to filter any more. Or? (But I observe that there are Lib.PG.Q.Values that I can use for filtering even if the supplied library doesn't have any q-value column) How's that?

**Reply by vdemichev** [reply]
Strange... Aren't those lib q-values all 0?

**Reply by JM-Bader** [reply]
yes, indeed - they are all 0. (I guess you impute them to 0). I guess they should rather be NaN, correct?

**Reply by JM-Bader** [reply]
so that means I need to filter based on Global.PG.Q.Value < 1% ?

**Reply by JM-Bader** [reply]
Additionally, I now don't understand how you envisioned this: [in the combined lib, delete all columns with q-values. Run search C with MBR. Filter by Lib.PG.Q.Value at 1%.] in the first place. Would I be supposed to match back the Lib.PG.Q.Values from the merged lib that I initially removed for the search?

**Reply by vdemichev** [reply]
MBR creates a new lib which has its own lib q-values

**Reply by JM-Bader** [reply]
so I need to load the *NEW lib file additional to the main report tsv and filter based on the Lib.PG.Q.Value there? (instead of the 0s in the main report)?

**Reply by JM-Bader** [reply]
FYI: in that resulting library, there is no Lib.PG.Q.Value named column but the PG.Q.Value one: 
Values are not all 0 there
![image](https://user-images.githubusercontent.com/98888664/218213790-f97e92af-c1d7-43bf-b13c-295661e9dc8e.png)

**Reply by vdemichev** [reply]
No, just filter the output report

**Reply by vdemichev** [reply]
Yes, libs never contain ‘lib’ q-value columns, only the main report
